components/reddit/sources/new-links-by-user/new-links-by-user.js

```diff
@@ -0,0 +1,78 @@
+const common = require("../common");
+const { reddit } = common.props;
+
+module.exports = {
+  ...common,
+  key: "new-links-by-user",
+  name: "New links by user",
+  description: "Emits an event each time a user posts a new link.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    username: { propDefinition: [common.props.reddit, "username"] },
+    numberOfParents: {
+      type: "integer",
+      label: "Number of parents",
+      description:
+        "The emitted events will contain the new comment plus the parents of said comment up to the number indicated in this property.",
+      default: 2,
+      min: 2,
+      max: 10
+    },
+    timeFilter: { propDefinition: [reddit, "timeFilter"] },
```

Comment:
I'd recommend making this prop optional with a default value of `all` (can be overridden here or just set to be optional with the default in the app file).

Reply:
I set this optional with the default value in the app file.